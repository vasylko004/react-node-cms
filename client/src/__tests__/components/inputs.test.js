import React from 'react';
import renderer from 'react-test-renderer';
import TextField from '../../components/inputs/text-filed';
import CheckBox from '../../components/inputs/checkbox';

test("Input renders testing ", ()=>{
   let component = renderer.create(<TextField 
        classes="test" 
        placeholder="test" 
        helperText="test" 
        type="text"
        name="test"
        />)

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(<TextField 
        classes="test" 
        placeholder="test" 
        helperText="test" 
        type="text"
        name="test"
        errorMessage = "test"
        isInvalid = {true}
        />)
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(<TextField 
        classes="test" 
        placeholder="test" 
        helperText="test" 
        type="text"
        name="test"
        disabled = {true}
        />)

    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
})

test("Checkbox input renders testing ", ()=>{
    let component = renderer.create(<CheckBox classes="test" name="test" label="test" />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.checked = true;
    expect(tree).toMatchSnapshot();
})