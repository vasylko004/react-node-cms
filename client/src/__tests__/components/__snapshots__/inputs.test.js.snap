// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox input renders testing  1`] = `
<div
  className="check-box"
>
  <label
    htmlFor="check_test"
  >
    <input
      defaultChecked={false}
      id="check_test"
      name="test"
      type="checkbox"
    />
    <span>
       
      test
       
    </span>
  </label>
</div>
`;

exports[`Checkbox input renders testing  2`] = `
<div
  checked={true}
  className="check-box"
>
  <label
    htmlFor="check_test"
  >
    <input
      defaultChecked={false}
      id="check_test"
      name="test"
      type="checkbox"
    />
    <span>
       
      test
       
    </span>
  </label>
</div>
`;

exports[`Input renders testing  1`] = `
<div
  className="input-field test"
>
  <input
    autoComplete=""
    className=""
    name="test"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="test"
    type="text"
  />
  <label
    className="active"
    onClick={[Function]}
  />
  <span
    className="helper-text"
    data-error="error"
  >
    test
  </span>
</div>
`;

exports[`Input renders testing  2`] = `
<div
  className="input-field test"
>
  <input
    autoComplete=""
    className="validate invalid"
    name="test"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="test"
    type="text"
  />
  <label
    className="active"
    onClick={[Function]}
  />
  <span
    className="helper-text"
    data-error="test"
  >
    test
  </span>
</div>
`;

exports[`Input renders testing  3`] = `
<div
  className="input-field test"
>
  <input
    autoComplete=""
    className=""
    disabled="disabled"
    name="test"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="test"
    type="text"
  />
  <label
    className="active"
    onClick={[Function]}
  />
  <span
    className="helper-text"
    data-error="error"
  >
    test
  </span>
</div>
`;
